plugins/coverage/token.js

```diff
@@ -37,27 +37,33 @@ module.exports = config => ({
                 tokenConfig.username = username;
             }
 
-            let data;
-            // Get job scope
-
+            // Get scope and job name
             if (jobId && !scope) {
                 const job = await jobFactory.get(jobId);
 
                 if (!job) {
                     throw boom.notFound(`Job ${jobId} does not exist`);
                 }
 
+                tokenConfig.jobName = job.name;
                 tokenConfig.scope =
                     job.permutations[0] && job.permutations[0].annotations
                         ? job.permutations[0].annotations[COVERAGE_SCOPE_ANNOTATION]
                         : null;
+            }
 
-                data = await config.coveragePlugin.getAccessToken(tokenConfig);
+            // Get pipeline name
+            if (pipelineId && (!projectName || projectName.includes('undefined'))) {
```

Comment:
is projectName an array? should we make it plural?

Reply:
No, it's just a string
sometimes it's set to `undefined:undefined`, so just trying to catch that
https://www.w3schools.com/jsref/jsref_includes.asp